lib/sumo/client.rb

```diff
@@ -79,8 +110,12 @@ def creds
   end
   private :creds
 
-  def connection
-    @connection ||= Excon.new('https://api.sumologic.com')
+  def connection(endpoint = nil)
```

Comment:
How about making a `@connections` instance variable, that holds a hash of HTTP endpoints to their Excon connections. I don't see why we would want to create a new connection for every request.